packages/fxa-content-server/app/scripts/models/account.js

```diff
@@ -293,6 +295,37 @@ const Account = Backbone.Model.extend(
       );
     },
 
+    /**
+     * Given a user's `kB` and a pipeline public key chosen at random from the
+     * key bundle, derive the ecosystem_user_id and encrypt it with the
+     * pipeline key, yielding the ecosystem_anon_id. Return the
+     * ecosystem_anon_id.
+     *
+     * @param {String} kB - user's kB
+     * @param {String} pipelineKey - a randomly-chosen pipeline public key
+     * in JWK format
+     * @returns {Promise} A promise that will be fulfilled with the user's
+     * ecosystem_anon_id in JWK format
+     */
+    async generateEcosystemAnonID(kB, pipelineKey) {
+      const uid = this.get('uid');
+      const keyData = {
+        // This identifier comes from the 'Account Ecosysten Telemetry
+        // Identifiers' google doc.
+        identifier: 'https://identity.mozilla.com/ids/ecosystem_telemetry',
+        // Following fxa-auth-server/lib/oauth/routes/key_data.js#L14-L20, FxA
+        // doesn't yet support per-user scoped key rotation; use a block of
+        // zeros instead.
+        keyRotationSecret:
+          '0000000000000000000000000000000000000000000000000000000000000000',
+        keyRotationTimestamp: 1595287165358,
+      };
```

Comment:
Ideally you would obtain these values from the server, but as a starting point that doesn't seem worth an extra request. I also expect that you don't actually use the `keyRotationTimestamp` value in this specific case, and it may be worth setting it to an obviously-fake value like `1000000000000`.

Reply:
Interestingly it does get passed into the HMAC call through [here](https://github.com/mozilla/fxa/blob/main/packages/fxa-content-server/app/scripts/lib/crypto/scoped-keys.js#L32-L33), we just don't do anything to vary it between users

Reply:
The other occurrence of this `keyRotationTimestamp` has a millisecond timestamp dating to when the library was first written back in 2017, which is why I went with yesterday's date for the timestamp 🤷